crates/starknet-devnet/tests/test_blocks_generation.rs

```diff
@@ -632,4 +632,33 @@ mod blocks_generation_tests {
 
         assert_get_class_hash_at(&devnet).await;
     }
+
+    #[tokio::test]
+    async fn get_data_by_specifying_latest_block_hash_and_number() {
+        let devnet = BackgroundDevnet::spawn().await.unwrap();
+        let (_, account_address) = devnet.get_first_predeployed_account().await;
+        let latest_block = devnet.get_latest_block_with_tx_hashes().await.unwrap();
+        let block_ids =
+            [BlockId::Hash(latest_block.block_hash), BlockId::Number(latest_block.block_number)];
+
+        for block_id in &block_ids {
+            let nonce = devnet.json_rpc_client.get_nonce(block_id, account_address).await.unwrap();
+            assert_eq!(nonce, FieldElement::ZERO);
+
+            let class_hash =
+                devnet.json_rpc_client.get_class_hash_at(block_id, account_address).await.unwrap();
+            assert_eq!(
+                class_hash,
+                FieldElement::from_hex_be(CAIRO_1_ACCOUNT_CONTRACT_SIERRA_HASH).unwrap()
+            );
+
+            let key = FieldElement::ZERO;
```

Comment:
Instead of testing with a dummy key, I'd test with something that returns a non-zero value, e.g. the account's public key, as we did in this test:
https://github.com/0xSpaceShard/starknet-devnet-rs/blob/6771dd84b7fdca82fdfdb43a0b5a1a7978d4e95e/crates/starknet-devnet/tests/test_fork.rs#L460

Reply:
913971493a677c978470ac51a527906f0235edd3

Reply:
Instead of relying on the `PREDEPLOYED_ACCOUNT_PUBLIC_KEY` constant, I'd use the `signer` returned by `get_first_predeployed_account` and call its `get_public_key` method, like we do in the example linked in my previous comment.

Reply:
267468a248bb603da6b64723e203ab0d3fbb2787